src/Form/DeliveryImportType.php

```diff
@@ -58,12 +61,29 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
             try {
 
-                $deliveries = $this->spreadsheetParser->parse($file->getPathname());
+                $result = $this->spreadsheetParser->parse($file->getPathname());
 
-                foreach ($deliveries as $delivery) {
+                $deliveries = array_filter($result->getData(), function($delivery, $rowNumber) use ($result) {
                     $violations = $this->validator->validate($delivery);
                     if (count($violations) > 0) {
-                        throw new \Exception((string) $violations->get(0));
+                        $result->addErrorToRow($rowNumber, (string) $violations->get(0));
```

Comment:
There is never more than one violation error ?

Reply:
Yes, you're right. I'll change it so we take into account all the errors no just the first one.